docs/src/pages/components/buttons/CustomizedButtons.js

```diff
@@ -66,6 +66,66 @@ const theme = createMuiTheme({
   palette: {
     primary: green,
   },
+  additions: {
+    MuiButton: {
+      dashed: {
+        padding: '5px 15px',
+        border: '5px dashed red',
+        '&$disabled': {
+          border: `5px dashed red`,
+        },
+      },
+      tertiery: {
+        backgroundColor: 'yellow',
+        padding: 20,
+      },
+    },
+  },
+  variants: {
+    MuiButton: [
+      {
+        trigger: (props) => props.variant === 'red',
+        styles: {
+          color: 'red'
+        }
+      },
+      {
+        trigger: (props) => props.size === 'xxsmall',
+        styles: {
+          fontSize: '5px'
+        }
+      },
+      {
+        trigger: (props) => props.toggle,
+        styles: {
+          background: 'green'
+        }
+      },
+      {
+        trigger: (props) => props.toggle === 'red',
+        styles: {
+          backgroundColor: 'red'
+        }
+      }
+    ],
+  },
+  variantsV2: {
```

Comment:
@oliviertassinari what do you think of this alternative?

Reply:
I find it better! I think that it's elegant, it finds a nice balance between the different constraints. Because it's not a function, we can display all the states in a generated documentation.